extensions/data-workspace/src/services/workspaceService.ts

```diff
@@ -54,12 +54,19 @@ export class WorkspaceService implements IWorkspaceService {
 
 		// create workspace
 		const projectFolder = vscode.Uri.file(path.dirname(projectFileFsPath));
-
-		if (isCurrentWorkspaceUntitled()) {
-			vscode.workspace.updateWorkspaceFolders(vscode.workspace.workspaceFolders!.length, null, { uri: projectFolder });
-			await getAzdataApi()?.workspace.saveAndEnterWorkspace(workspaceFile!);
+		const azdataApi = getAzdataApi();
+		if (azdataApi) {
+			if (isCurrentWorkspaceUntitled()) {
+				vscode.workspace.updateWorkspaceFolders(vscode.workspace.workspaceFolders!.length, null, { uri: projectFolder });
+				await azdataApi.workspace.saveAndEnterWorkspace(workspaceFile!);
+			} else {
+				await azdataApi.workspace.createAndEnterWorkspace(projectFolder, workspaceFile);
+			}
 		} else {
-			await getAzdataApi()?.workspace.createAndEnterWorkspace(projectFolder, workspaceFile);
+			// In VS Code we don't have access to the workspace APIs exposed by ADS and so can't actually create a new saved workspace.
+			// Instead we'll just always call this, which will either add it to the existing untitled workspace or create a new
+			// untitled workspace which the user can then save later on as they wish.
+			vscode.workspace.updateWorkspaceFolders(vscode.workspace.workspaceFolders?.length || 0, null, { uri: projectFolder });
```

Comment:
the temp project shouldn't be saved to the global state on line 53 if a new workspace isn't getting created and entered, otherwise it'll get added again the next time VS Code restarts. Since VS Code isn't getting restarted when `vscode.workspace.updateWorkspaceFolders() `is called, the project will be added to the workspace file later in `addProjectsToWorkspace()`.